src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -450,6 +462,107 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
     }
   }
 
+  /**
+    * Collect data from Blocks Table, Operations Table, and the Tezos RPC,
+    * return the appropriate list of VoteAggregates, and write them to the Votes table.
+    * @param ex the needed ExecutionContext to combine multiple database operations
+    * @return a future result of the number of rows stored to db, if supported by the driver
+    */
+  private[this] def processTezosVotes(): Future[Done] = {
+    import slick.jdbc.PostgresProfile.api._
+    implicit val mat = ActorMaterializer()
+
+    logger.info("Processing latest Tezos votes data...")
+
+    /**
+      * * Helper function, takes a sequence of GovernanceData, separates them by ballot types.
+      * @param votingData
+      * @return
+      */
+    def partitionByVote(votingData: Seq[VotingData]): (Seq[VotingData], Seq[VotingData], Seq[VotingData]) = {
+      val (yays, notYays) = votingData.partition(x => x.ballot.contains("yay"))
+      val (nays, notYaysOrNays) = notYays.partition(x => x.ballot.contains("nay"))
+      val (passes, _) = notYaysOrNays.partition(x => x.ballot.contains("pass"))
+      (yays, nays, passes)
+    }
+
+    /**
+      * * Helper function, takes a sequence of GovernanceData, calculates stake.
+      * @param votingData
+      * @return
+      */
+    def calculateStake(votingData: Seq[VotingData]): Future[BigDecimal] =
+      Future
+        .sequence(
+          votingData.map {
+            case VotingData(_, _, hash, _, _, _, address) =>
+              tezosNodeOperator.getAccountBalanceForBlock(BlockHash(hash), AccountId(address.get))
```

Comment:
What's `address` here? What if `.get` fails?

Reply:
that's the part about which I'm guessing that the vote should always have address, but it was fetched from the operations table, but there address is optional

Reply:
I see